server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala

```diff
@@ -53,6 +61,14 @@ case object ReceivedAtSortProperty extends SortProperty {
   override def toSortClause: SortClause = SortClause.Arrival
 }
 
+case object AllInThreadHaveKeywordSortProperty extends SortProperty {
+  override def toSortClause: SortClause = throw UnsupportedSortException("allInThreadHaveKeyword")
```

Comment:
Modify the trait to return an `Either[UnsupportedSortException, SortClause]` and handle the exception upstream 

(don't throw, any scalaist will tell you throwing is a side effect, and that it is not compatible with functional programming)